shamir/shamir.go

```diff
@@ -0,0 +1,260 @@
+package shamir
```

Comment:
This is forked from [Vault](https://github.com/hashicorp/vault/tree/master/shamir) and does not need to be reviewed.

Reply:
I don't think we can because we don't want to vendor the entire `vault` codebase, which is fairly massive.